content/backend/graphql-js/4-adding-a-database.md

```diff
@@ -7,189 +7,235 @@ answers: ["To increase performance of the server", "When using two APIs, the Gra
+Let's see it in action with your project!
 
 <Instruction>
 
-Open `datamodel.prisma` and add the following code:
+Open `schema.prisma` and add the following code:
 
-```graphql{2,3}(path=".../hackernews-node/prisma/datamodel.prisma")
-type Link {
-  id: ID! @id
-  createdAt: DateTime! @createdAt
-  description: String!
-  url: String!
+```graphql{2,3}(path=".../hackernews-node/prisma/schema.prisma")
+// 1
+datasource db {
+  provider = "sqlite" 
+  url      = "file:./dev.db"
+}
+
+// 2
+generator client {
+  provider = "prisma-client-js"
+}
+
+// 3
+model Link {
+  id          Int      @id @default(autoincrement())
+  createdAt   DateTime @default(now())
+  description String
+  url         String
 }
 ```
 
 </Instruction>
 
-There are two main differences compared to the previous `Link` version from `schema.graphql`.
+Let's break down the three parts:
 
-First, you're adding the `@id` directive to the `id: ID!` field. This means Prisma will auto-generate and store globally unique IDs for the `Link` records in the database on the `id` field.
+1. **Data source**: Tells Prisma you'll be using SQLite for your database connection.
+1. **Generator**: Indicates that you want to genenerate Prisma Client. 
+1. **Data model**: Here, we have written out our `Link` as a model.
 
-Second, you're adding a new field called `createdAt: DateTime! @createdAt`. Thanks to the `@createdAt` directive, this field is also managed by Prisma and will be read-only in the API. It stores the time for when a specific `Link` was created. You can also annotate a field with the `@updatedAt` directive to track when a record was last updated.
+You may notice that this `Link` *model* looks very similar to the `Link` *type* in the GraphQL Schema. It's important to quickly discuss how the two schemas are different.
 
-Now, let's see what you need to do with `prisma.yml`.
+// explain how they are different
 
-<Instruction>
 
-Add the following contents to `prisma.yml`:
+It's finally time to actually create our SQLite database, which Prisma can do for us right out of the box with a simple command!
 
-```graphql(path=".../hackernews-node/prisma/prisma.yml")
-# The HTTP endpoint for your Prisma API
-endpoint: ''
+<Instruction>
 
-# Points to the file that contains your datamodel
-datamodel: datamodel.prisma
+From the root directory of your project, create your first *migration* by running the following command in your terminal:
 
-# Specifies language & location for the generated Prisma client
-generate:
-  - generator: javascript-client
-    output: ../src/generated/prisma-client
+```bash(path=".../hackernews-node/")
+npx prisma migrate save --experimental
 ```
 
+You will get a prompt asking if you would like to create a new database. Select "Yes", and type "Init DB" for the `Name of migration`.
```

Comment:
I believe the `<Instruction>` component is too limited for this, it can only have: 

1. normal text
1. code block

In this case you have:

1. normal text
1. code block
1. normal text

Which I think breaks it iirc, best factor the prompt into itsown `<Instruction>` block.

Reply:
ah! Good catch, I wasn't aware of the limitation. Just refactored it and will keep that in mind moving forward 🙂